1.1.1 bugfix release

asmeurer opened the pull request:
Looks like a lot of little issues have popped up with 1.1, so I'm going a 1.1.1 bugfix release. 

This branch is based on the 1.1 branch, so if you want any fixes here, please be sure they are made against this branch, not master.

jirikuncar committed:
Fix Quantity Add postprocessor

Fixes issue with sorted dictionaries in Python 3.6.

jirikuncar committed:
Test Quantity Add postprocessor

matthew-brett committed:
Allow for complex Numpy scalars in sympify

As of Sympy 1.1, the following raises a recursion error:

>>> c = np.longcomplex(1 + 1j)
>>> print(sympify(c))

This is because the sympify code had the following:

   if not isinstance(a, np.floating):
       return sympify(np.asscalar(a))

In the case of `c` above, it is not `np.floating` type (it's complex)
and `np.asscalar(c)` returns another numpy scalar (rather than a Python
scalar, as would be the case for the `np.complex` type), so causing the
loop to go into infinite recursion.

Add handling of the complex case directly in the sympify clause above,
and add test for `np.longcomplex` type.

While I'm at it, add test for `np.longdouble` type, which will be
`np.float96` on 32-bit platforms, `np.float128` on 64-bit.

asmeurer committed:
Fix various recursion issues in Add assumptions

These only show up when running with the cache off, as otherwise the caching
prevents infinite recursion.

Unfortunately, I don't know how to test this, aside from running the test
suite with the cache off (which we should be doing on Travis or somewhere but
aren't). There are also a lot of other assumedly unrelated test failures
with ./bin/test -C at the moment, although you can see an improvement as
'./bin/test -C core' crashes without this patch and doesn't with it.

Fixes #12907.

asmeurer committed:
Fix a test failure on 32-bit systems

See #12902. Doesn't fix the other failure reported there.

asmeurer committed:
Update version for 1.1.1rc1

asmeurer committed:
Merge remote-tracking branch 'jirikuncar/patch-1' into 1.1.1

asmeurer committed:
Merge pull request #12917 from matthew-brett/fix-sympy-longcomplex-1.1.1

MRG: Allow for complex Numpy scalars in sympify

jirikuncar committed:
Test _constructor_postprocessor_mapping on subclasses

jirikuncar committed:
Lookup postprocessors for subclasses

jirikuncar committed:
Update language classifiers -2.6 +3.6

jirikuncar committed:
Removal of PyTest from postprocessor test

asmeurer committed:
Merge pull request #12923 from jirikuncar/setup-classifiers

Update language classifiers -2.6 +3.6

asmeurer committed:
Merge pull request #12919 from jirikuncar/postprocess_subclasses

Lookup postprocessors for subclasses

jirikuncar committed:
Test addition of a quantity and a number

jirikuncar committed:
Fix addition of a quantity and a number

closes #12918

asmeurer committed:
Only call asarray in lambdify on Python builtin numeric types

That way it doesn't break things for custom array subtypes.

I'm not sure how to test this.

Fixes #12940.

ixjlyons committed:
Fix using preprocessor_statements kwarg in CCodeGen.

jirikuncar committed:
Workaround power of a quantity

asmeurer committed:
Merge pull request #12948 from ixjlyons/fix_preproc_stmt_kwarg

Fix using preprocessor_statements kwarg in CCodeGen.

siefkenj committed:
Fix stacking of empty matrices

siefkenj committed:
Fix for #12950

asmeurer committed:
Merge pull request #12952 from jirikuncar/12918-quantity-and-addition

Fix addition of a quantity and a number

jksuom commented on sympy/core/power.py:
```diff
@@ -1062,15 +1062,22 @@ def _eval_is_rational(self):
             return True
         elif self.exp.is_rational:
             if self.base.is_algebraic is False:
                 return self.exp.is_nonzero
```
This should be `is_zero`.

asmeurer replied:
I believe this is unrelated to the above fix (let me know if I am wrong). If so, we should fix this in a PR to master.

asmeurer committed:
Remove unused import in numpydoc.py

It causes an error with the development version of Sphinx.

asmeurer committed:
Suppress citation warnings in Sphinx

asmeurer committed:
Suppress footnote warnings

For some reason some of the references in the solveset docs are being seen as
unused footnotes.

asmeurer committed:
Remove make html-errors from docs build

Now make html always errors on warnings.

siefkenj committed:
Added test for issue #12950

asmeurer committed:
Build the docs with the latest version of Sphinx and docutils

This should fix #12958 (search is broken in the docs).

asmeurer committed:
Install latexmk in Travis for building the docs

siefkenj committed:
Fix spelling error

asmeurer committed:
Fix sed command to work on Mac

asmeurer committed:
Fix latexmk to use xetex in docs build

asmeurer committed:
Fix footnote in Heaviside docstring

Sphinx treats all footnotes as references (if otherwise, I don't know how to
make it work). The * was breaking the latex build.

asmeurer committed:
Merge pull request #12946 from asmeurer/asarray-lambdify

Only call asarray in lambdify on Python builtin numeric types

asmeurer committed:
Fix sed command in docs Makefile

jirikuncar committed:
Support functions with Quantities and Dimensions

closes #12980

Co-authored-by: Stan Schymanski <schymans@gmail.com>

jirikuncar committed:
Use self.func when evaluating Abs(Quantity)

jirikuncar commented:
Please merge regularly maintenance branches to master:

```
master
------
o (v1.1.0)
| \
o   o maint-1.1
|   | ---------
o   o
| / |  <- regular merge
o   o (v1.1.1)
| / |  <- regular merge
o   o
|   |
o   o (v1.1.2)
| / |
o   .
.   .
.   .

asmeurer commented:
Sorry, I got caught up with SciPy. I am hoping to finish this up this week.

asmeurer committed:
Merge pull request #12957 from siefkenj/fixes1.1.1

Fix Matrix Regressions

asmeurer committed:
Merge pull request #12909 from asmeurer/32-bit

Fix a test failure on 32-bit systems

asmeurer committed:
Use dist: trusty in Travis

This will hopefully fix the LaTeX build issues. Also, it will eventually be
the default anyway
(https://blog.travis-ci.com/2017-07-11-trusty-as-default-linux-is-coming).

asmeurer committed:
Remove checkout and pull from Dockerfile

The file to do that is now in master.

asmeurer committed:
Don't rebuild the Docker image in release.sh

Instead, whenever the Dockerfile is changed, it will need to be rebuilt
manually. The latest image should always be pushed to DockerHub, so that you
can just run the release script and it will pull it down without having to
build it.

asmeurer committed:
Update AUTHORS file

asmeurer committed:
Run the docker container with -t (makes Ctrl-C work)

asmeurer committed:
Enable tracing in rever to see where things are hanging

asmeurer committed:
Merge pull request #12965 from asmeurer/sphinx-update

Build the docs with the latest version of Sphinx

jirikuncar committed:
Fix prefix definitions

jirikuncar commented:
> Sorry, I got caught up with SciPy. I am hoping to finish this up this week.

The release is not so important as long as all commits from 1.1.1 are immediately merged to master so we don't have to manually merge or cherry-picked fixes which appear in both branches.

The releases should happen independently on merges between maintenance branches. See a real life example from another project:

```
*   3e246af0e Merge branch 'maint-1.2' into legacy
|\
| *   48c62330b Merge branch 'maint-1.1' into maint-1.2
| |\
| | *   89b4eee2a Merge branch 'maint-1.0' into maint-1.1
| | |\
| | | * 75373d2c2 installation: support for Ubuntu 14.04 LTS
| | | * cc5779c16 installation: use of `/usr/sbin/service`
| | | * 4585ef594 WebSession: silencing MySQL UTF-8 warnings
| * | | 5a7481ac7 installation: Apache and Redis start-up
| * | | 6a9cf772f WebSearch: fix for da today 3650 test case
| * | | 1c2818e96 BibCatalog: no email tests when no SMTP server
| * | | 23c9c67dc installation: web node h5py provisioning
* | | |   cebdae899 Merge branch 'maint-1.2' into legacy
|\ \ \ \
| |/ / /
| * | |   7e02ddc5f Merge branch 'maint-1.1' into maint-1.2
| |\ \ \
| | |/ /
| | * | d8fc81d9b installation: Invenio hosting of jQuery plugins
| | * |   aa0513494 Merge branch 'maint-1.0' into maint-1.1
| | |\ \
```

asmeurer committed:
Replace /home with /root in the release scripts

Since /root is the actual ~ in the Docker container.

asmeurer committed:
Merge branch '1.1.1' of github.com:sympy/sympy into 1.1.1

asmeurer committed:
Merge pull request #13009 from jirikuncar/prefix-definitions

Fix prefix definitions

asmeurer committed:
Merge pull request #12990 from jirikuncar/12980

Support functions with Quantities and Dimensions

asmeurer committed:
Better logic for Add assumptions recursion

When the monotonic_sign stuff would lead to a recursion (only happens when the
cache if off, see #12907, #12908), instead of returning None, skip to the
other logic later in the _eval_is_assumption routine.

asmeurer committed:
Install latexmk in the release Dockerfile

asmeurer committed:
Merge branch '1.1.1' of github.com:sympy/sympy into 1.1.1

asmeurer committed:
Fix /home -> /root in release.sh

asmeurer committed:
Merge pull request #12908 from asmeurer/assumptions-recursion

Fix various recursion issues in Add assumptions

asmeurer committed:
Set the version in the rever file

This should work with newer versions of rever.

asmeurer committed:
Fix rever commands in the release README

siefkenj committed:
Fix matrix stacking for sparse matrices

siefkenj committed:
Merge remote-tracking branch 'upstream/1.1.1' into fixes1.1.1

asmeurer committed:
Merge pull request #13031 from siefkenj/fixes1.1.1

Fix matrix stacking for sparse matrices

asmeurer committed:
Bump version to 1.1.1 (final)

asmeurer committed:
Comment out code in rever.xsh that assumes the tag isn't a merge commit

asmeurer committed:
Add missing update_websites activity to rever file

asmeurer committed:
Merge branch 'master' into 1.1.1

asmeurer committed:
Merge branch 'master' into 1.1.1

asmeurer committed:
Fix units tests
